index.js

```diff
@@ -155,6 +156,22 @@ function fastifyWebsocket (fastify, opts, next) {
     let isWebsocketRoute = false
     let wsHandler = routeOptions.wsHandler
     let handler = routeOptions.handler
+    const handleUpgrade = routeOptions.handleUpgradeRequest
+      ? (request, reply, callback) => {
+          const rawRequest = request.raw
+          routeOptions.handleUpgradeRequest(request, rawRequest[kWs], rawRequest[kWsHead])
+            .then(socket => {
+              callback(socket)
```

Comment:
```suggestion
              process.nextTick(callback, socket)
```

Otherwise any error throw by the callback above would be swallowed by the following catch block